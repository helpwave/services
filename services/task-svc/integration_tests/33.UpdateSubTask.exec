#!/bin/bash

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask

echo "empty update"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f2e70feb-9137-44d7-bcad-b0e90bd0706a"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/UpdateSubTask

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask


echo "update"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87", "name": "Hi c:"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/UpdateSubTask

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/GetTask

grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87", "name": "Repair of retinal tear by photocoagulation of unspecified type"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/UpdateSubTask

echo "Does not exists:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InRlc3RpbmUudGVzdEBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"id": "42e70feb-9137-44d7-bcad-b0e90bd0706b", "name": "Hi c:"}' \
	localhost:3001 \
	proto.services.task_svc.v1.TaskService/UpdateSubTask
