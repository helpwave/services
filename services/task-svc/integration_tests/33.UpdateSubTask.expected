empty update
{}
update
{}
Does not exists:
{
  "id": "f531ae7e-96cd-4d5d-ab62-1afbe90fdd87",
  "name": "LDR Brachytherapy of Thorax Lymph using Palladium 103",
  "description": "Restriction of Accessory Pancreatic Duct with Intraluminal Device, Via Natural or Artificial Opening",
  "status": "TASK_STATUS_TODO",
  "assignedUserId": "df363387-7774-4592-b01c-2b765abaa86a",
  "patient": {
    "id": "ddb6c139-cd0d-4c65-95d8-288035eabf54",
    "name": "Byram Zealy"
  },
  "public": true,
  "dueAt": "2024-03-19T06:33:32Z",
  "createdBy": "531244c4-0500-4701-91ef-a03645ba92b2",
  "subtasks": [
    {
      "id": "7be17800-a372-4b8a-aa92-05b6cdff5178",
      "name": "Local excision of lesion or tissue of bone, patella",
      "done": true,
      "createdBy": "46697884-4792-49ff-b243-14a17eb5b871"
    },
    {
      "id": "f2e70feb-9137-44d7-bcad-b0e90bd0706a",
      "name": "Hi c:",
      "createdBy": "73c0d265-3c9a-410a-8eaf-0174fc57f421"
    }
  ],
  "organizationId": "3b25c6f5-4705-4074-9fc6-a50c28eba406"
}
